lib/components/Input/base/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { QuillSvgProps } from "@deriv/quill-icons";
+import clsx from "clsx";
+import {
+    ForwardRefExoticComponent,
+    HTMLInputTypeAttribute,
+    InputHTMLAttributes,
+    forwardRef,
+    useState,
+} from "react";
+import "./base.scss";
+import React from "react";
+import { TMediumSizes } from "../../../types";
+
+export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+    type?: HTMLInputTypeAttribute;
+    icon?: ForwardRefExoticComponent<Omit<QuillSvgProps, "ref">>;
+    statusIcon?: ForwardRefExoticComponent<Omit<QuillSvgProps, "ref">>;
+    inputSize?: TMediumSizes;
+    status?: "neutral" | "success" | "error";
+    disabled?: boolean;
+    variant?: "fill" | "outline";
+    leftStatusMessage?: string;
+    rightStatusMessage?: string;
+    textAlignment?: "left" | "center";
+    label?: string;
+    value?: string;
+}
+
+const IconSize = {
+    sm: { width: 24, height: 24 },
+    md: { width: 24, height: 24 },
+};
+
+const statusIconColors = {
+    neutral: "status-icon--neutral",
+    success: "status-icon--success",
+    error: "status-icon--error",
+};
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+    (
+        {
+            type = "text",
+            inputSize = "md",
+            className,
+            status = "neutral",
+            disabled = false,
+            variant = "outline",
+            icon: Icon,
+            leftStatusMessage,
+            rightStatusMessage,
+            textAlignment = "left",
+            label,
+            statusIcon: StatusIcon,
+            onChange,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [hasValue, setHasValue] = useState(false);
+
+        return (
+            <div className="quill-input__container">
+                <div
+                    data-has-value={hasValue}
+                    className={clsx(
+                        className,
+                        `quill-input__wrapper`,
+                        `quill-input__wrapper__variant--${variant}`,
+                        variant === "fill" && `status--${status}`,
+                        `quill-input__wrapper__size--${inputSize}`,
+                        `quill-input__wrapper__status--${status}`,
+                    )}
+                >
+                    {Icon && (
+                        <div>
+                            <Icon {...IconSize[inputSize]} />
+                        </div>
+                    )}
+                    <input
+                        {...rest}
+                        type={type}
```

Comment:
Should we restrict some types? What if type='checkbox' will be passed?
I think we should work with type='text'

Reply:
yes, for now I'm allowing `text`, `email`, `password`